### ISSUE TITLE
Isolate the UI code in a separate library

### ISSUE THREAD

**Issue by amol-mandhane** [emacs-lsp/lsp-ui#50]
LSP UI has UI components which can be reused by non-LSP packages in many different contexts.

For example,
Sideline: Can be used as ElDoc/Flycheck front-end.
Peek: XRef front-end.

Since lsp-mode already integrates with above packages, it could be great if the UI part of this package can be released separately.

**Comment by sebastiencs** [comment]
Yes I have thought about that.
But for now I prefer to work on a single repo, once the components get mature maybe...
A library to manipulate child frames would be nice too

**Comment by topisani** [comment]
yeah, i thought about this as well, id like to use especially the docs child frame and xref-peek - i think waiting a bit makes sense

**Comment by dgutov** [comment]
+1 on `lsp-ui-peek` especially. I'd be happy to use it with other backends. And in GNU ELPA especially.

Sideline for ElDoc sounds intriguing as well, although it might be difficult to reconcile with the existing API.

**Comment by brotzeit** [comment]
Hopefully somebody finds the motivation to do this.

**Comment by brotzeit** [comment]
@dgutov Sorry for closing this issue. I think I was a little too motivated while trying to clean the issue tracker. Please add a comment in the future if you think an issue should be kept open. I'm not even sure if I closed the issue intentionally :D

**Comment by dgutov** [comment]
@brotzeit Hey, no problem.

Thing is, for this to happen, there needs to be both a concerted agreement between the users who wrong the code, and someone motivated to do the extraction. I don't have spare time for this personally, so commenting "please reopen" didn't seem appropriate. It's not something I can demand anyway.

**Comment by brotzeit** [comment]
Even if you don't have the time to do it yourself, it's absolutely legitimate to say the issue should be left open for others who might have the time to do it.
